### PR TITLE
Remove snapshots and auditlog from setup and portal folders

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses

This Pull Request adds the `IDoNotSupportSnapshots` interface to setup and portal folders. Otherwise, a snapshot is created each time an object is added (e.g. inside `setup/departments`)

## Current behavior before PR

A snapshot of the folder is created each time an object is added inside

## Desired behavior after PR is merged

No snapshots are created and "Audit Log" for setup and portal folders is not displayed

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/senaite/core/setuphandlers.py [hunk 1]
```diff
@@ -221,6 +227,10 @@ def add_dexterity_setup_items(portal):
     setup = api.get_setup()
     add_dexterity_items(setup, items)
 
+    # disable snapshots for setup folders
+    objects = [setup.get(item[0]) for item in items]
+    map(disable_snapshots, objects)
```
Comment (ramonski) [inline src/senaite/core/setuphandlers.py]: Why is this required when the base class already implements `IDoNotSupportSnapshots`?

File: src/senaite/core/setuphandlers.py [hunk 2]
```diff
@@ -250,6 +260,26 @@ def add_senaite_setup_items(portal):
...
+    alsoProvides(obj, IDoNotSupportSnapshots)
+
+    # do not display audit log
+    noLongerProvides(obj, IAuditable)
+
+    # remove all snapshots
+    annotation = IAnnotations(obj)
+    storage = annotation.get(snapshot.SNAPSHOT_STORAGE)
+    if storage:
+        del(annotation[snapshot.SNAPSHOT_STORAGE])
```
Comment (ramonski) [inline src/senaite/core/setuphandlers.py]: Please create a function in the snapshot API to remove this.

File: src/senaite/core/upgrade/v02_06_000.py [hunk 3]
```diff
@@ -265,3 +266,31 @@ def update_workflow_role_mappings(wf_id, objs_or_brains):
...
+    # remove non-objects
+    folders = filter(api.is_object, folders)
+    folders = list(set(folders))
+
+    # remove setup folders (they hold settings as fields)
+    skip = [bika_setup, setup]
+    folders = filter(lambda folder: folder not in skip, folders)
+
+    # disable auditlog and snapshots
+    map(disable_snapshots, folders)
```
Comment (ramonski) [inline src/senaite/core/upgrade/v02_06_000.py:294]: Better use an explicit API function instead using the the one from `setuphandlers`

**Review by ramonski — CHANGES_REQUESTED** [review]

File: src/senaite/core/setuphandlers.py (hunk 1 above)
Comment (xispa) [reply src/senaite/core/setuphandlers.py]: Good point

Comment (xispa) [reply src/senaite/core/setuphandlers.py]: Removed with https://github.com/senaite/senaite.core/pull/2479/commits/94ff2a92c1be84c8c433bad3e53216c150044de4

File: src/senaite/core/setuphandlers.py (hunk 2 above)
Comment (xispa) [reply src/senaite/core/setuphandlers.py]: Done with https://github.com/senaite/senaite.core/pull/2479/commits/94ff2a92c1be84c8c433bad3e53216c150044de4

File: src/senaite/core/upgrade/v02_06_000.py (hunk 3 above)
Comment (xispa) [reply src/senaite/core/upgrade/v02_06_000.py:294]: Done with https://github.com/senaite/senaite.core/pull/2479/commits/94ff2a92c1be84c8c433bad3e53216c150044de4

**Review by ramonski — APPROVED** [review]
